compiler/qsc/src/interpret.rs

```diff
@@ -560,14 +580,21 @@ impl Interpreter {
         Ok(circuit)
     }
 
+    /// Sets the entry expression for the interpreter.
+    pub fn set_entry_expr(&mut self, entry_expr: &str) -> std::result::Result<(), Vec<Error>> {
+        let (graph, _) = self.compile_entry_expr(entry_expr)?;
+        self.expr_graph = Some(graph);
+        Ok(())
+    }
+
     /// Runs the given entry expression on the given simulator with a new instance of the environment
     /// but using the current compilation.
     pub fn run_with_sim(
         &mut self,
         sim: &mut impl Backend<ResultType = impl Into<val::Result>>,
         receiver: &mut impl Receiver,
         expr: Option<&str>,
-    ) -> std::result::Result<InterpretResult, Vec<Error>> {
+    ) -> InterpretResult {
```

Comment:
Nice catch! This extra layer of indirection was leftover from several refactors ago, glad you finally eliminated it.